highlighting/lib/src/highlightv2.dart

```diff
@@ -406,4 +406,72 @@ class HighlightV2 {
 
     return emitter;
   }
+
+  Result highlightAuto(
+    String code,
+    List<String> languageSubset,
+  ) {
+    final plainText = justTextHighlightResult(code);
+    try {
+      final results = languageSubset
+          .where((e) => _languages[e] != null || builtinLanguages[e] != null)
+          .where((e) =>
+              _languages[e]?.disableAutodetect != true ||
+              builtinLanguages[e]?.disableAutodetect != true)
+          .map((name) => highlight(name, code, false))
+          .toList();
+
+      results.insert(0, plainText);
+      results.sortByCompare<Result>((element) => element, _resultComparator);
+
+      return results[0];
+    } on Exception {
+      return plainText;
+    }
+  }
+
+  Result justTextHighlightResult(String code) {
+    final emitter = Result(
+      language: null,
+      top: Mode(
+        name: 'text-plain',
+        disableAutodetect: true,
+      ),
+    );
+
+    return emitter..addText(code);
+  }
+
+  int _resultComparator(Result a, Result b) {
+    // sort base on relevance
+    if ((a.relevance - b.relevance).abs() > 0.0001) {
+      if (a.relevance < b.relevance) {
+        return 1;
+      }
+      if (a.relevance > b.relevance) {
+        return -1;
+      }
+    }
+
+    // always award the tie to the base language
+    // ie if C++ and Arduino are tied, it's more likely to be C++
+    if (a.language != null && b.language != null) {
+      if (_getLanguage(a.language!)?.supersetOf == b.language) {
+        return 1;
+      }
+      if (_getLanguage(b.language!)?.supersetOf == a.language) {
+        return -1;
+      }
+    }
+
+    // otherwise say they are equal, which has the effect of sorting on
+    // relevance while preserving the original ordering - which is how ties
+    // have historically been settled, ie the language that comes first always
+    // wins in the case of a tie
```

Comment:
```suggestion
    // Otherwise say they are equal, which has the effect of sorting on
    // relevance while preserving the original ordering - which is how ties
    // have historically been settled, i.e. the language that comes first always
    // wins in the case of a tie.
```